backend/tests/v5/unit/middleware/teamspaces/projects/containers/containers.test.js

```diff
@@ -0,0 +1,310 @@
+/**
+ *  Copyright (C) 2021 3D Repo Ltd
+ *
+ *  This program is free software: you can redistribute it and/or modify
+ *  it under the terms of the GNU Affero General Public License as
+ *  published by the Free Software Foundation, either version 3 of the
+ *  License, or (at your option) any later version.
+ *
+ *  This program is distributed in the hope that it will be useful,
+ *  but WITHOUT ANY WARRANTY; without even the implied warranty of
+ *  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ *  GNU Affero General Public License for more details.
+ *
+ *  You should have received a copy of the GNU Affero General Public License
+ *  along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+const { src } = require('../../../../../helper/path');
+
+jest.mock('../../../../../../../src/v5/utils/responder');
+const Responder = require(`${src}/utils/responder`);
+const { templates } = require(`${src}/utils/responseCodes`);
+
+const Containers = require(`${src}/middleware/teamspaces/projects/containers/containers`);
+const { stringToUUID } = require(`${src}/utils/helper/uuids`);
+
+// Mock respond function to just return the resCode
+Responder.respond.mockImplementation((req, res, errCode) => errCode);
+
+const testFilterContainerData = () => {
+	describe('Filter container data', () => {
+		test('next() should be called if request body is empty', () => {
+			const mockCB = jest.fn(() => {});
+			Containers.filterContainerData({}, {}, mockCB);
+			expect(mockCB.mock.calls.length).toBe(1);
+		});
+
+		test('next() should be called with whitelist fields retained', async () => {
+			const mockCB = jest.fn(() => {});
+			const body = {
+				name: 'container name',
+				unit: 'mm',
+				desc: 'this is a container',
+				type: 'Structure',
+				defaultView: '7591fbdb-52b9-490a-8a77-fdb57c57dbc8',
+				defaultLegend: '120965d4-dd6e-4505-a2d2-e9a5cdcaad81',
+				subModels: [],
+				surveyPoints: [],
+				angleFromNorth: [],
+				elevation: 10,
+			};
+			const expectedResult = { ...body };
+			Containers.filterContainerData({ body }, {}, mockCB);
+			expect(mockCB.mock.calls.length).toBe(1);
+			expect(body).toEqual(expectedResult);
+		});
+
+		test('next() should be called with un-whitelisted fields removed', async () => {
```

Comment:
just throw it back with invalid arguments.

we want to go strict on unexpected fields